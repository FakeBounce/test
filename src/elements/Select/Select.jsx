import React, { Component } from "react";
import styles from "./Select.module.scss";
import { Input } from "reactstrap";
import { observer } from "mobx-react";
import { observable } from "mobx";
import { translate } from "common/methods/translations";

@observer
class SelectElement extends Component {
  @observable selectedElement = "";

  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    const { initialOption } = this.props;
    this.selectedElement = initialOption || "initial";
  }

  handleChange(event) {
    this.selectedElement = event.target.value;
    const { selectedOption } = this.props;
    selectedOption(this.selectedElement);
  }

  static mapOptions(allOptions) {
    return allOptions.map((singleOption, index) => {
      return (
        <option
          key={`options-${singleOption.name}-${index}`}
          name={singleOption.name}
          value={singleOption.value}
        >
          {singleOption.name}
        </option>
      );
    });
  }

  render() {
    const { setOptions, className } = this.props;
    return (
      <div className={`${styles.Container} ${className}`}>
        <Input
          type="select"
          name="select"
          value={this.selectedElement}
          onChange={this.handleChange}
        >
          <option value="">{translate("globals.selectOption")}</option>
          {setOptions && setOptions.length > 0 ? (
            SelectElement.mapOptions(setOptions)
          ) : (
            <div />
          )}
        </Input>
      </div>
    );
  }
}

export default SelectElement;
