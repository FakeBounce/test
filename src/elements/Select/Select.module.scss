.Container {
  height: 100%;
  div {
    &:global(.form-group) {
      margin-bottom: 0;
      select {
        padding: 12px 20px;
        height: initial;
        box-shadow: 0 2px 4px 0 rgba(0,0,0,0.12);
        border-radius: 2px;
        border: 0;
      }
    }
  }
}