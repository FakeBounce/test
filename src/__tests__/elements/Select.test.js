import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import Select from "elements/Select/Select";

configure({ adapter: new Adapter() });
const selectedOption = jest.fn();

describe("Testing Select component", () => {
  it("renders as expected without props", () => {
    const wrapper = shallow(<Select />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders as expected with props", () => {
    const wrapper = shallow(
      <Select setOptions={[{ name: "1", value: 1 }, { name: "2", value: 2 }]} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("functions behave correctly", () => {
    const wrapper = shallow(<Select selectedOption={selectedOption} />);
    wrapper.instance().handleChange({ target: { value: "test" } });
    expect(wrapper.instance().selectedElement).toBe("test");
    expect(selectedOption.mock.calls.length).toBe(1);
  });
});
