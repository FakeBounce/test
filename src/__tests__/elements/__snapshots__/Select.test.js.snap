// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Select component renders as expected with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectElement
    setOptions={
        Array [
            Object {
              "name": "1",
              "value": 1,
            },
            Object {
              "name": "2",
              "value": 2,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Input
        name="select"
        onChange={[Function]}
        type="select"
        value=""
>
        <option
                value=""
        >
                Sélectionner une option
        </option>
        <option
                name="1"
                value={1}
        >
                1
        </option>
        <option
                name="2"
                value={2}
        >
                2
        </option>
</Input>,
      "className": "undefined undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <option
            value=""
>
            Sélectionner une option
</option>,
          Array [
            <option
              name="1"
              value={1}
>
              1
</option>,
            <option
              name="2"
              value={2}
>
              2
</option>,
          ],
        ],
        "name": "select",
        "onChange": [Function],
        "type": "select",
        "value": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sélectionner une option",
            "value": "",
          },
          "ref": null,
          "rendered": "Sélectionner une option",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "options-1-0",
          "nodeType": "host",
          "props": Object {
            "children": "1",
            "name": "1",
            "value": 1,
          },
          "ref": null,
          "rendered": "1",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "options-2-1",
          "nodeType": "host",
          "props": Object {
            "children": "2",
            "name": "2",
            "value": 2,
          },
          "ref": null,
          "rendered": "2",
          "type": "option",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Input
          name="select"
          onChange={[Function]}
          type="select"
          value=""
>
          <option
                    value=""
          >
                    Sélectionner une option
          </option>
          <option
                    name="1"
                    value={1}
          >
                    1
          </option>
          <option
                    name="2"
                    value={2}
          >
                    2
          </option>
</Input>,
        "className": "undefined undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <option
              value=""
>
              Sélectionner une option
</option>,
            Array [
              <option
                name="1"
                value={1}
>
                1
</option>,
              <option
                name="2"
                value={2}
>
                2
</option>,
            ],
          ],
          "name": "select",
          "onChange": [Function],
          "type": "select",
          "value": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sélectionner une option",
              "value": "",
            },
            "ref": null,
            "rendered": "Sélectionner une option",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "options-1-0",
            "nodeType": "host",
            "props": Object {
              "children": "1",
              "name": "1",
              "value": 1,
            },
            "ref": null,
            "rendered": "1",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "options-2-1",
            "nodeType": "host",
            "props": Object {
              "children": "2",
              "name": "2",
              "value": 2,
            },
            "ref": null,
            "rendered": "2",
            "type": "option",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Testing Select component renders as expected without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectElement />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Input
        name="select"
        onChange={[Function]}
        type="select"
        value=""
>
        <option
                value=""
        >
                Sélectionner une option
        </option>
        <div />
</Input>,
      "className": "undefined undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <option
            value=""
>
            Sélectionner une option
</option>,
          <div />,
        ],
        "name": "select",
        "onChange": [Function],
        "type": "select",
        "value": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sélectionner une option",
            "value": "",
          },
          "ref": null,
          "rendered": "Sélectionner une option",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Input
          name="select"
          onChange={[Function]}
          type="select"
          value=""
>
          <option
                    value=""
          >
                    Sélectionner une option
          </option>
          <div />
</Input>,
        "className": "undefined undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <option
              value=""
>
              Sélectionner une option
</option>,
            <div />,
          ],
          "name": "select",
          "onChange": [Function],
          "type": "select",
          "value": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sélectionner une option",
              "value": "",
            },
            "ref": null,
            "rendered": "Sélectionner une option",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
